### DISCUSSION TITLE
[文章加密] 彻底解决notion-id被用户获取的问题

### DISCUSSION THREAD

**Discussion by zs6** [tangly1024/NotionNext#2081] [Ideas]
**描述bug**
可以找到notion原始地址并获取密码

**复现步骤**
通过搜索role":"reader" 可以获取notion id，如果博客用户名和notion用户名一致，便可以拼出noiton原始地址，从原始地址就可以获取密码

**期望的正常结果**
隐藏notion id

**环境**

- 操作系统: windows10
- 浏览器 edge110
- NotionNext版本 3.16.4
- 主题 所有主题

**补充说明**
无

**Comment by tangly1024** [comment]
已收到反馈

**Comment by tangly1024** [comment]
有点麻烦：
页面渲染的话就必须要拿到notion的id，而前端一旦拿到id，理论上就能直接访问notion页面，而notion页面又是共享的。

根治方案：
除非用一种第三方介入，notion页面不共享到网页上，即使任何人拿到此id也无法访问原notion页面，
但用户自己可以仅授权给此第三方notion页面的访问权限。

【最终方案】
后续可能考虑创建一个notion的官方授权应用来解决此问题

**Comment by yutaolian** [comment]
> 有点麻烦： 页面渲染的话就必须要拿到notion的id，而前端一旦拿到id，理论上就能直接访问notion页面，而notion页面又是共享的。
> 
> 根治方案： 除非用一种第三方介入，notion页面不共享到网页上，即使任何人拿到此id也无法访问原notion页面， 但用户自己可以仅授权给此第三方notion页面的访问权限。
> 
> 【最终方案】 后续可能考虑创建一个notion的官方授权应用来解决此问题

你好，我觉的这个问题是因为你把notion的页面共享出去的原因导致的,我看了您的操作步骤。有一步是你获得database_id的时候你把这个文档发布出去了。你可以使用这个按钮
<img width="595" alt="image" src="https://github.com/tangly1024/NotionNext/assets/6284480/14a576c7-e387-4277-86fa-175e668fadc2">
文档的database_id地址也能拿到。notion api的js-sdk也能访问到数据，就不会有这个密码泄露的问题了。另外我看您使用的是notion-clent的一个库，我看了一下notion 官方文档，可以Create an integration 然后和具体的database_id绑定。您可以看一下这个文章https://developers.notion.com/docs/create-a-notion-integration 这样就能解决文件密码可能会泄露的问题了

**Reply by cnjoby** [reply]
这个需要notionnext更新代码么？还是我们自己去建一个notion的intergration就可以了？

**Comment by Angel0726** [comment]
插眼

**Comment by siuze** [comment]
有关加密文章的密码安全性问题可有任何进展吗？之前似乎也讨论过是否可以设置密码为一个环境变量名，通过把密码定义到环境变量下来改善可能会被用户获取的这个问题。不知现在有没有合适的解决方案了

**Comment by BerBai** [comment]
notion官方提供integration授权访问，应该可以改成这种方式获取文章

**Comment by nickdlkk** [comment]
建议密码使用MD5之类的摘要，防止通过原始分享页面获取明文密码
